src/charm.py

```diff
@@ -313,18 +316,31 @@ def _mimir_version(self) -> Optional[str]:
         version_output, _ = self._container.exec(["/bin/mimir", "-version"]).wait_output()
         # Output looks like this:
         # Mimir, version 2.4.0 (branch: HEAD, revision: 32137ee)
-        result = search("Mimir, version {} ", version_output)
-
-        if result is None:
-            return result
-
-        return result[0]
+        if result := re.search(r"[Vv]ersion:?\s*(\S+)", version_output):
```

Comment:
Before you ask @rbarry82, this is speculative future proofing balanced with readability :sunglasses:

Reply:
Regular expressions solve all string problems 🥼

Reply:
Many, many years ago an old friend told: 

> "If you have an issue to solve and use regular expressions, now you have two"

:stuck_out_tongue: